p2p/pex/pex_reactor.go

```diff
@@ -297,6 +296,15 @@ func (r *PEXReactor) ReceiveAddrs(addrs []*p2p.NetAddress, src Peer) error {
 
 		err := r.book.AddAddress(netAddr, srcAddr)
 		r.logErrAddrBook(err)
+
+		// If this address came from a seed node, try to connect to it without waiting.
+		// TODO: Change this to an O(log(N)) check on a sorted seedAddr list
+		seedAddrs, _ := p2p.NewNetAddressStrings(r.config.Seeds)
```

Comment:
As seeds are not changing throughout the lifecycle of the process wouldn't it make sense to create a lookup (`map[NetAddress]struct{}`) once and use that instead.

Reply:
Great point!

Reply:
I'll change it to this for now, but on further thought I'm worried that the constant overhead of a hash function may make this less efficient in the expected case of a few seeds. But this is definitely on the order of things we can optimize post launch!

Reply:
Which hash function are you referring to?

Reply:
the internal hash function which will be used in the hashmap. (Namely its `memhash` here on 64 bit systems: https://golang.org/src/runtime/hash64.go)

Reply:
I don't think this is something we need to optimise for, even post-launch.

Reply:
Just ran the golang source runtime benchmarks. Slightly skewed speeds on my system, since it has the very nice hardware AES hash :)
```
BenchmarkHash5-8                          	200000000	         9.12 ns/op	 548.11 MB/s
BenchmarkHash16-8                         	200000000	         9.59 ns/op	1668.44 MB/s
BenchmarkHash64-8                         	100000000	        13.4 ns/op	4777.66 MB/s
BenchmarkHash1024-8                       	20000000	        76.0 ns/op	13467.29 MB/s
BenchmarkHash65536-8                      	  300000	      4340 ns/op	15097.94 MB/s
```
For a small number of seeds, a linear scan would probably be faster. But I'll switch to the hashmap here anyway, optimizing linear / logarithmic scans vs hashmaps is something we can defo do later. I'll write this up into an issue though, since it seems useful